src/GafferScene/EvaluateLightLinks.cpp

```diff
@@ -70,7 +70,7 @@ void EvaluateLightLinks::affects( const Gaffer::Plug *input, AffectedPlugsContai
 {
 	SceneProcessor::affects( input, outputs );
 
-	if( input == inPlug()->attributesPlug() || input == inPlug()->setPlug() || input == inPlug()->globalsPlug() )
+	if( SetAlgo::affectsSetExpression( input ) )
```

Comment:
We should keep the attributesPlug check here (and take it out of affectsSetExpression), and also check that `plug->parent() == inPlug()` before calling `affectsSetExpression()`.